src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = findIndex(hash(key));
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> iterator = table[index];
+            while (iterator.next != null || Objects.equals(iterator.key, key)) {
+                if (Objects.equals(iterator.key, key)) {
+                    iterator.value = newNode.value;
+                    return;
+                }
+                iterator = iterator.next;
+            }
+            iterator.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = findIndex(hash(key));
+        Node<K, V> iterator = table[index];
+        while (iterator != null) {
+            if (Objects.equals(iterator.key, key)) {
+                return iterator.value;
+            }
+            iterator = iterator.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkSize() {
+        if (size == threshold) {
+            resize();
+        }
+    }
+
+    private void resize() {
+        int capacity = table.length << 1;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        Node<K, V>[] temporaryTable = table;
+        table = (Node<K, V>[]) new Node[capacity];
+        size = 0;
+        transfer(temporaryTable);
+    }
+
+    private void transfer(Node<K, V>[] temporaryTable) {
+        for (Node<K, V> element : temporaryTable) {
+            while (element != null) {
+                put(element.key, element.value);
+                element = element.next;
+            }
+        }
+    }
+
+    private int findIndex(int hashValue) {
+        return Math.abs(hashValue & table.length - 1);
```

Comment:
why do you use such a formula, and not `%`?

Reply:
This formula do the same work, but is better comparing performance. I tried both of them, just when we use % we need to use Math.abs(hashValue % table.length);